### PR TITLE
Patch call for DDoS Protection plan

### PR DESCRIPTION
**Pull request by mancgupta**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/ddosProtectionPlan.json [hunk 1]
```diff
@@ -180,6 +180,58 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "Update successful. The operation returns the resulting DDoS protection plan resource.",
+            "schema": {
+              "$ref": "#/definitions/DdosProtectionPlan"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
```
Comment (sanjaiganesh) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/ddosProtectionPlan.json]: >"x-ms-long-running-operatio [](start = 8, length = 27)

should have 201, similar to PUT, as it is marked as long running operation.. ('201 + provisioningState' pattern)

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/ddosProtectionPlan.json [hunk 2]
```diff
@@ -180,6 +180,58 @@
...
+        "responses": {
+          "200": {
+            "description": "Update successful. The operation returns the resulting DDoS protection plan resource.",
+            "schema": {
+              "$ref": "#/definitions/DdosProtectionPlan"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
```
Comment (sanjaiganesh) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/ddosProtectionPlan.json]: same here.. 201

Comment (mancgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/ddosProtectionPlan.json]: Only changes in the Tags are supported with PATCH call. That's why, there won't be 201 for this call.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/ddosProtectionPlan.json (hunk 1 above)
Comment (mancgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/ddosProtectionPlan.json]: Only changes in the Tags are supported with PATCH call. That's why, there won't be 201 for this call.

**Comment by mancgupta** [comment]
@sanjaiganesh I have addressed the comments. Please review again.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/ddosProtectionPlan.json (hunk 2 above)
Comment (sanjaiganesh) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/ddosProtectionPlan.json]: In such a case, can you remove  "x-ms-long-running-operation": true, or set it to false

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/ddosProtectionPlan.json (hunk 1 above)
Comment (sanjaiganesh) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/ddosProtectionPlan.json]: In such a case, can you remove  "x-ms-long-running-operation": true, or set it to false

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/ddosProtectionPlan.json (hunk 2 above)
Comment (mancgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/ddosProtectionPlan.json]: Changed

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/ddosProtectionPlan.json (hunk 1 above)
Comment (mancgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/ddosProtectionPlan.json]: changed to false

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/ddosProtectionPlan.json [hunk 3]
```diff
@@ -180,6 +180,58 @@
...
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Update successful. The operation returns the resulting DDoS protection plan resource.",
```
Comment (sanjaiganesh) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/ddosProtectionPlan.json]: >The operation returns the resulting DDoS protection plan resource. [](start = 47, length = 66)

remove this text as it is not LRO anymore

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/ddosProtectionPlan.json [hunk 4]
```diff
@@ -180,6 +180,58 @@
             "$ref": "./examples/DdosProtectionPlanCreate.json"
           }
         }
+      },
+      "patch": {
+        "tags": [
+          "DdosProtectionPlans"
+        ],
+        "operationId": "DdosProtectionPlans_UpdateTags",
+        "description": "Update a DDoS protection plan tags",
```
Comment (sanjaiganesh) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/ddosProtectionPlan.json]: update both of these to remove 'tags'

Comment (mancgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/ddosProtectionPlan.json]: Done.

**Comment by MikhailTryakhov** [comment]
closed due to milestone was missed. please retarget to newer branch when it's ready

**Pull request closed without merging**